packages/zowe-explorer/src/job/actions.ts

```diff
@@ -464,3 +464,74 @@ async function deleteMultipleJobs(jobs: ReadonlyArray<IZoweJobTreeNode>, jobsPro
         await errorHandling(userMessage);
     }
 }
+
+export async function cancelJobs(jobsProvider: IZoweTree<IZoweJobTreeNode>, nodes: IZoweJobTreeNode[]): Promise<void> {
+    if (!nodes.length) {
+        return;
+    }
+
+    // Filter out nodes that have already been cancelled
+    const filteredNodes = nodes.filter(
+        (n) => n.job == null || n.job.retcode == null || !(n.job.retcode.includes("CANCEL") || n.job.retcode?.includes("ABEND"))
+    );
+    if (!filteredNodes.length) {
+        await Gui.showMessage(localize("cancelJobs.alreadyCancelled", "The selected jobs were already cancelled."));
+        return;
+    }
+
+    const jesApis = {};
+
+    const failedJobs: { job: zowe.IJob; error: string }[] = [];
+    // Build list of common sessions from node selection
+    const sessionNodes = [];
+    for (const jobNode of nodes) {
+        if (!jobNode.job) {
+            continue;
+        }
+        const sesNode = jobNode.getSessionNode();
+        const sesLabel = sesNode.label as string;
+        if (!(sesLabel in jesApis)) {
+            jesApis[sesLabel] = ZoweExplorerApiRegister.getJesApi(sesNode.getProfile());
+        }
+
+        if (!jesApis[sesLabel].cancelJob) {
+            failedJobs.push({
+                job: jobNode.job,
+                error: localize("cancelJobs.notImplemented", "The cancel function is not implemented in this API."),
+            });
+            continue;
+        }
+
+        try {
+            const cancelled = await jesApis[sesLabel].cancelJob(jobNode.job);
```

Comment:
I wasn't sure what the best approach was for this. But, I thought of the scenario where a user would select job nodes from multiple sessions, and perform the "Cancel Job" action... 

- If we immediately display an error that the operation is not supported, it might give the end-user the impression that the action failed completely. However, some jobs might have still cancelled successfully with the scenario mentioned.
- If we show a dialog _after_ the action is finished (showing what jobs failed to cancel), the user has to wait for the action to fail, but it will highlight exactly what jobs failed to cancel. Jobs that didn't fail to cancel will not show in the dialog.

Maybe I should adjust this dialog so that it will also list the jobs that cancelled successfully? This way, the user is aware of what jobs failed and what jobs succeeded when cancelling.